smoke-test.sh

```diff
@@ -236,18 +243,18 @@ function setupDevCerts() {
 function runAllTests() {
     # Run tests for each language and template
     if [ "$excludeNonWebTests" == "false" ]; then
-        doCommand C# console new restore build run
-        doCommand C# classlib new restore build
+        doCommand C# console new restore build run "publish --self-contained false" "publish --self-contained true -r linux-x64" "publish --self-contained true -r $targetRid"
+        doCommand C# classlib new restore build "publish --self-contained false" "publish --self-contained true -r linux-x64" "publish --self-contained true -r $targetRid"
```

Comment:
These are already just aliases rather than the full set of commands we run per test case, so could we just call it `publish` here and trifurcate in a conditional branch in `doCommand`?

Should get rid of the need to specially handle spaces and dashes?